tech/post/2017-09-23-Running-the-kubernetes-on-OpenStack-step-by-step.md

```diff
@@ -0,0 +1,601 @@
+
+This guide will take you through the steps of deploying Kubernetes to OpenStack manually. Firstly we will install a devstack environment, and then create a Nova instance in this devstack. We will compile and install a single node kubernetes environment in this Nova instance.
+
+## Install a devstack environment
+
+DevStack is a series of extensible scripts used to quickly bring up a complete OpenStack environment based on the latest versions of everything from git master. It is used interactively as a development environment and as the basis for much of the OpenStack project’s functional testing. So here we will use devstack to install a complete OpenStack environment for running the kubernetes.
+
+
+### 1 The environment
+
+OS: Ubuntu 16.04 
+CPU: Intel(R) Xeon(R) CPU E5-2407 0 @ 2.20GHz
+Memory: 32G
+Storage: 100G
+
+### 2 Download DevStack
+Devstack should be run as a non-root user with sudo enabled (standard logins to cloud images such as "ubuntu" or "cloud-user" are usually fine).
+You can quickly create a separate stack user to run DevStack with:
+```bash
+$ sudo useradd -s /bin/bash -d /opt/stack -m stack
+```
+Since this user will be making many changes to your system, it should have sudo privileges:
+```bash
+$ echo "stack ALL=(ALL) NOPASSWD: ALL" | sudo tee /etc/sudoers.d/stack
+$ sudo su - stack
+```
+Download DevStack:
+```bash
+$ git clone https://git.openstack.org/openstack-dev/devstack
+$ cd devstack
+```
+The devstack repo contains a script that installs OpenStack and templates for configuration files.
+
+### 3 Create a local.conf and Install 
+```bash
+[[local|localrc]]
+
+DATABASE_PASSWORD=password
+RABBIT_PASSWORD=password
+SERVICE_PASSWORD=password
+SERVICE_TOKEN=password
+ADMIN_PASSWORD=password
+
+enable_plugin karbor http://git.trystack.cn/openstack/karbor master
+enable_plugin karbor-dashboard http://git.trystack.cn/openstack/karbor-dashboard master
+enable_plugin heat http://git.trystack.cn/openstack/heat master
+
+#run the services you want to use
+ENABLED_SERVICES=rabbit,mysql,key
+ENABLED_SERVICES+=,n-cpu,n-api,n-obj,n-cond,n-sch,n-novnc,n-cauth,n-api-meta
+ENABLED_SERVICES+=,placement-api,placement-client
+ENABLED_SERVICES+=,neutron,q-svc,q-agt,q-dhcp,q-l3,q-meta,q-metering
+ENABLED_SERVICES+=,cinder,g-api,g-reg
+ENABLED_SERVICES+=,c-api,c-vol,c-sch,c-bak,horizon
+ENABLED_SERVICES+=,heat,h-api,h-api-cfn,h-api-cw,h-eng
+
+#Add the karbor services
+enable_service karbor-api
+enable_service karbor-operationengine
+enable_service karbor-protection
+
+#Add the karbor-dashboard services
+enable_service karbor-dashboard
+
+#disable the default services you don't want to use
+disable_service n-net
+disable_service n-cell
+
+SWIFT_HASH=66a3d6b56c1f479c8b4e70ab5c2000f5
+SWIFT_REPLICAS=1
+SWIFT_DATA_DIR=$DEST/data
+enable_service s-proxy s-object s-container s-account
+
+
+```
+The project karbor is a data protection orchestration as a service in OpenStack. It will be installed in this devstack. 
```

Comment:
The Karbor is not related to this blog :)

Reply:
This blog is prepared for the environment about portecting the k8s pod with karbor.